autoload/go/guru.vim

```diff
@@ -198,23 +197,26 @@ function! s:async_guru(args) abort
   endfunction
 
   let start_options = {
-        \ 'close_cb': function("s:close_cb"),
+        \ 'close_cb': funcref("s:close_cb"),
+        \ 'in_mode': 'raw',
+        \ 'out_mode': 'nl',
+        \ 'err_mode': 'nl',
         \ }
 
-  if &modified
-    let l:tmpname = tempname()
-    call writefile(split(result.stdin_content, "\n"), l:tmpname, "b")
-    let l:start_options.in_io = "file"
-    let l:start_options.in_name = l:tmpname
-  endif
-
```

Comment:
I think we're using it it for `async_guru` as well. The command is the same for both `sync_guru` and `async_guru`. They both get it from `guru_cmd`. So if it's modified, it uses the `-modified` flag. If you also check `def.vim`, you'll see that we're also using it there as well for `guru`. I just want the `stdin` a part of the `start_options`. However right now you changed, we start the job first, and then we send some other content to it. I think that would cause a race condition. I'm ok with your other changes, but I'm not sure this one is needed.